plutus-core/untyped-plutus-core/test/Transform/Simplify.hs

```diff
@@ -1,206 +1,179 @@
+Note that the @(int integer 1)@ won't get inlined: it isn't pre-inlined because
 @x@ occurs twice, and it isn't post-inlined because @costIsAcceptable Constant{} = False@.
 However, the entire term will be inlined since it is pure.
 -}
 inlinePure3 :: Term Name PLC.DefaultUni PLC.DefaultFun ()
-inlinePure3 = mkInlinePurityTest $ do
-  x <- freshName "x"
-  y <- freshName "y"
-  pure $
-    Apply
-      ()
-      (LamAbs () x $ LamAbs () y $ Apply () (Var () x) (Var () x))
-      (mkConstant @Integer () 1)
-
-{- | @force ([(\x -> delay (\y -> [x x])) (delay ([error (con integer 1)]))])@ is pure,
+inlinePure3 = mkInlinePurityTest $ lam x (lam y (var x `app` var x)) `app` int 1
+ where
+  (x, y) = uniqueNames2 "x" "y"
+
+{-| @force ([(\x -> delay (\y -> [x x])) (delay ([error (int integer 1)]))])@ is pure,
 but it is very tricky to see so. It requires us to match up a force and a
 delay through several steps of intervening computation.
 -}
 inlinePure4 :: Term Name PLC.DefaultUni PLC.DefaultFun ()
-inlinePure4 = mkInlinePurityTest $ do
-  x <- freshName "x"
-  y <- freshName "y"
-  pure . Force () $
-    Apply
-      ()
-      (LamAbs () x $ Delay () $ LamAbs () y $ Apply () (Var () x) (Var () x))
-      (Delay () $ Apply () (Error ()) $ mkConstant @Integer () 1)
+inlinePure4 =
+  mkInlinePurityTest $
+    force $
+      lam x (delay (lam y (var x `app` var x)))
+        `app` delay (app err (int 1))
+ where
+  (x, y) = uniqueNames2 "x" "y"
 
 -- | @error@ is impure.
 inlineImpure1 :: Term Name PLC.DefaultUni PLC.DefaultFun ()
-inlineImpure1 = mkInlinePurityTest $ pure $ Error ()
+inlineImpure1 = mkInlinePurityTest err
 
 -- | @force (delay error)@ is impure.
 inlineImpure2 :: Term Name PLC.DefaultUni PLC.DefaultFun ()
-inlineImpure2 = mkInlinePurityTest $ pure . Force () . Delay () $ Error ()
+inlineImpure2 = mkInlinePurityTest $ force $ delay err
 
-{- | @force (force (force (delay (delay (delay (error))))))@ is impure, since it
+{-| @force (force (force (delay (delay (delay (error))))))@ is impure, since it
 is the same as @error@.
 -}
 inlineImpure3 :: Term Name PLC.DefaultUni PLC.DefaultFun ()
 inlineImpure3 =
-  mkInlinePurityTest
-    $ pure
-      . Force ()
-      . Force ()
-      . Force ()
-      . Delay ()
-      . Delay ()
-      . Delay ()
-    $ Error ()
-
-{- | @force (force (force (delay (delay a))))@ is impure, since @a@ may expand
+  mkInlinePurityTest $
+    force . force . force . delay . delay $
+      delay err
+
+{-| @force (force (force (delay (delay a))))@ is impure, since @a@ may expand
 to an impure term such as @error@.
 -}
 inlineImpure4 :: Term Name PLC.DefaultUni PLC.DefaultFun ()
 inlineImpure4 =
-  mkInlinePurityTest $
-    Force () . Force () . Force () . Delay () . Delay () . Var () <$> freshName "a"
+  mkInlinePurityTest $ force . force . force . delay . delay . var $ name "a" 0
 
-{- | @(\a -> f (a 0 1) (a 2)) (\x y -> g x y)@
+inlineImpure5 :: Term Name PLC.DefaultUni PLC.DefaultFun ()
+inlineImpure5 =
```

Comment:
Have you verified that before your change, the inliner won't inline this? I think that should be the case but it's worth verifying to be certain.

Reply:
Yes, I did.
```
inlineImpure5:           FAIL
  Test output was different from 'untyped-plutus-core/test/Transform/inlineImpure5.uplc.golden'. 
  It was:
  \a -> (\b -> addInteger (addInteger 1 a) b) (addInteger 3 a)
  Use -p '/simplify/&&/inlineImpure5/' to rerun this test only.
```